pulseapi/entries/models.py

```diff
@@ -56,24 +56,28 @@ def public(self):
             # and so if its query set is checked, it'll crash out due
             # to the absence of the associated ModerationState table.
             approved = ModerationState.objects.get(name='Approved')
-            return self.filter(
-                    moderation_state=approved
-                ).prefetch_related(
-                    'tags',
-                    'issues',
-                    'help_types',
-                    'published_by',
-                    'bookmarked_by',
-                    'bookmarked_by__profile__related_user',
-                    'published_by__profile',
-                    'moderation_state',
-                    'related_creators__creator__profile__related_user',
-                )
+            return self.filter(moderation_state=approved)
 
         except:
             print("could not make use of ModerationState!")
             return self.all()
 
+    def with_related(self):
+        """
+        Return all entries with their related data as separate queries
+        """
+
+        return self.prefetch_related(
```

Comment:
Why did you revert the change made in the prefetch?

Reply:
Oh I just split the queries into two chainable functions. The prefetch query is the same though. I needed one without the prefetch because it ran too many queries for a query in which I just needed moderated entries that were filtered.

Reply:
Oh nevermind, I see what you meant. Must be a rebase error, sorry, fixing!

Reply:
> Oh I just split the queries into two chainable functions. The prefetch query is the same though. I needed one without the prefetch because it ran too many queries for a query in which I just needed moderated entries that were filtered.

can you unpack this a bit? it's hard to follow

Reply:
We did two things in one function - we prefetched a bunch of stuff for entries AND we filtered based on moderation state. Since these are two completely unrelated tasks, I split them into two functions. Everywhere we expected both tasks to be run now instead call both these functions to run those tasks. The tasks are still run as a single query because of the way queryset chaining works. But, I had to get entries that were filtered based on moderation state but I didn't want it to prefetch all those related fields (tags, issues, etc.) because I didn't need them and needed to create a separate prefetch clause.

What @patjouk is referring to was that I think I made a rebase mistake and overrode her changes with the old code. I reverted that so that my change now is to only split one function into two functions without changing any of the queries.

Reply:
Got it, thanks!